File: server/models/note.js
const mongoose = require("mongoose");

const NoteSchema = new mongoose.Schema({
  student: {
    // Note for which student
    type: mongoose.Schema.Types.ObjectId,
    ref: "Student",
    required: true,
  },
  date: {
    type: Date,
    default: Date.now,
  },
  description: {
    type: String,
    required: true,
  },
  noteAdderType: {
    // Who added this note
    type: String,
    enum: ["PARENT", "DRIVER"],
    required: true,
  },
  driverPhone: {
    // Which driver will see this note.
    type: String,
    required: true,
  },
});

module.exports = Note = mongoose.model("Note", NoteSchema);

File: socoolbus/lib/models/note.dart
class Note {
  final String noteId;
  final String student;
  final String school;
  final String noteAdderType;
  final String description;

  const Note(
      {required this.noteId,
      required this.student,
      required this.school,
      required this.noteAdderType,
      required this.description});

  factory Note.fromJson(Map<dynamic, dynamic> json) {
    return Note(
      noteId: json['_id'],
      student: json['student'],
      school: json['school'],
      noteAdderType: json['noteAdderType'] == "DRIVER" ? "Şoför" : "Veli",
      description: json['description'],
    );
  }
}
